SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,169 +30,207 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
+
+		RPCArray.Add(RemoteFunction);
+		Info.RPCInfoMap.Add(RemoteFunction, RPCInfo);
 	}
-}
 
-void USpatialTypebindingManager::CreateTypebindings()
-{
-	// This will iterate through all supported classes: both Actor and Subobject classes
-	// For subobject classes, an FClassInfo will be created with RPC metadata and handover data
-	// but schema components will be 0 since subobject classes don't have associated components
-	// afterwards, the subobject FClassInfos will be the basis of the FClassInfo for specific subobjects
-	// and have their proper schema components filled in
-	for (UClass* Class : SupportedClasses)
+	for (TFieldIterator<UProperty> PropertyIt(Class); PropertyIt; ++PropertyIt)
 	{
-		FClassInfo Info;
+		UProperty* Property = *PropertyIt;
 
-		TArray<UFunction*> RelevantClassFunctions = improbable::GetClassRPCFunctions(Class);
-
-		for (UFunction* RemoteFunction : RelevantClassFunctions)
+		if (Property->PropertyFlags & CPF_Handover)
 		{
-			ESchemaComponentType RPCType = SCHEMA_Invalid;
-			if (RemoteFunction->FunctionFlags & FUNC_NetClient)
-			{
-				RPCType = SCHEMA_ClientRPC;
-			}
-			else if (RemoteFunction->FunctionFlags & FUNC_NetServer)
-			{
-				RPCType = SCHEMA_ServerRPC;
-			}
-			else if (RemoteFunction->FunctionFlags & FUNC_NetCrossServer)
-			{
-				RPCType = SCHEMA_CrossServerRPC;
-			}
-			else if (RemoteFunction->FunctionFlags & FUNC_NetMulticast)
-			{
-				RPCType = SCHEMA_NetMulticastRPC;
-			}
-			else
+			for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
 			{
-				checkNoEntry();
-			}
+				FHandoverPropertyInfo HandoverInfo;
+				HandoverInfo.Handle = Info.HandoverProperties.Num() + 1; // 1-based index
+				HandoverInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
+				HandoverInfo.ArrayIdx = ArrayIdx;
+				HandoverInfo.Property = Property;
 
-			TArray<UFunction*>& RPCArray = Info.RPCs.FindOrAdd(RPCType);
+				Info.HandoverProperties.Add(HandoverInfo);
+			}
+		}
 
-			FRPCInfo RPCInfo;
-			RPCInfo.Type = RPCType;
-			RPCInfo.Index = RPCArray.Num();
+		if (Property->PropertyFlags & CPF_AlwaysInterested)
+		{
+			for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
+			{
+				FInterestPropertyInfo InterestInfo;
+				InterestInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
+				InterestInfo.Property = Property;
 
-			RPCArray.Add(RemoteFunction);
-			Info.RPCInfoMap.Add(RemoteFunction, RPCInfo);
+				Info.InterestProperties.Add(InterestInfo);
+			}
 		}
+	}
 
-		for (TFieldIterator<UProperty> PropertyIt(Class); PropertyIt; ++PropertyIt)
+	ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
+	{
+		Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+		if (ComponentId != 0)
 		{
-			UProperty* Property = *PropertyIt;
+			Info.SchemaComponents[Type] = ComponentId;
+			ComponentToClassInfoMap.Emplace(ComponentId, MakeShared<FClassInfo>(Info));
```

Comment:
We're duplicating the `FClassInfo` here. What we should do instead is change `ClassInfoMap` to `TMap<UClass*, TSharedRef<FClassInfo>>` and then pass that `TSharedRef` here.
Note the usage of `TSharedRef` instead of `TSharedPtr`, indicating that it can't be null.

Reply:
Follow-up task https://improbableio.atlassian.net/browse/UNR-930 . Hopefully, I get a chance to do it.